portal/src/test/scala/cromwell/pipeline/controller/UserControllerTest.scala

```diff
@@ -0,0 +1,101 @@
+package cromwell.pipeline.controller
+
+import akka.http.scaladsl.model.ContentTypes.`application/json`
+import akka.http.scaladsl.model.{ HttpEntity, StatusCodes }
+import akka.http.scaladsl.testkit.ScalatestRouteTest
+import cromwell.pipeline.datastorage.dto.user.DeactivateUserRequestByEmail
+import cromwell.pipeline.datastorage.dto.{ UserDeactivationByEmailResponse, UserDeactivationByIdResponse, UserId }
+import cromwell.pipeline.service.UserService
+import cromwell.pipeline.utils.auth.SecurityDirective
+import cromwell.pipeline.{ AuthConfig, ExpirationTimeInSeconds }
+import org.scalamock.scalatest.MockFactory
+import org.scalatest.{ Matchers, WordSpec }
+import pdi.jwt.JwtAlgorithm
+import play.api.libs.json.Json
+
+import scala.concurrent.Future
+
+class UserControllerTest extends WordSpec with Matchers with MockFactory with ScalatestRouteTest {
+
+  private val UserService = stub[UserService]
+  private val UserController = new UserController(UserService)
+
+  "UserController" when {
+    "deactivateByEmail" should {
+      "return email and false value if user was successfully deactivated" in {
+        val deactivateUserRequestByEmail = DeactivateUserRequestByEmail("email")
+        val emailResponse = UserDeactivationByEmailResponse(email = "JohnDoe@cromwell.com", active = false)
+
+        val httpEntity = HttpEntity(`application/json`, Json.stringify(Json.toJson(deactivateUserRequestByEmail)))
```

Comment:
I guess we should work with project Entity (like `deactivateUserRequestByEmail`) and not with the `HttpEntity`. Marshalling/unmarshalling should be handled by marshallers/unmarshallers we wrote. You need to import them to do so.

Reply:
BTW, `deactivateUserRequestByEmail` is a strange name. Why not `deactivateUserByEmailRequest`?

Reply:
Agree, it looks more English in this way